SmartDeviceLink/private/SDLIAPDataSession.m

```diff
@@ -58,10 +58,14 @@ - (void)writeDataToSessionStream {
             if (bytesWritten >= 0) {
                 if (bytesWritten == bytesRemaining) {
                     [self.sendDataQueue popBuffer];
-                } else {
+                } else if (bytesRemaining > bytesWritten) {
```

Comment:
Is this more convenient?
```suggestion
                } else if (bytesRemaining >= bytesWritten) {
```

Reply:
I don't think the greater than equal conditions would make sense here, since the first if statement already does an equal comparison and we only want this part of the code to run when `bytesRemaining` is greater than `bytesWritten`.